controllers/admin/AdminThemesController.php

```diff
@@ -142,16 +142,24 @@ public function initPageHeaderToolbar()
         parent::initPageHeaderToolbar();
 
         if (empty($this->display)) {
-            $this->page_header_toolbar_btn['import_theme'] = array(
-                'href' => self::$currentIndex.'&action=importtheme&token='.$this->token,
-                'desc' => $this->trans('Add new theme', array(), 'Admin.Design.Feature'),
-                'icon' => 'process-icon-new'
-            );
-            $this->page_header_toolbar_btn['export_theme'] = array(
-                'href' => self::$currentIndex.'&action=exporttheme&token='.$this->token,
-                'desc' => $this->trans('Export current theme', array(), 'Admin.Design.Feature'),
-                'icon' => 'process-icon-export'
-            );
+            if (!in_array(
+                $this->authorizationLevel(),
+                array(
+                    AdminController::LEVEL_VIEW,
+                    AdminController::LEVEL_EDIT,
```

Comment:
I guess I see what you mean, but it would be better to check the sames permissions for displaying the buttons and for the actual actions.

You could make only one function for this use case:

```php
protected function isEditGranted()
{
   return in_array(
            $this->authorizationLevel(),
            array(
                AdminController::LEVEL_ADD,
                AdminController::LEVEL_DELETE,
            )
        );
}
```

Then, you call this function l.145, 254 & 597. By doing this, we will avoid issues if one day somebody needs to update the required permissions for these actions.